include/ruby/io.h

```diff
@@ -59,6 +59,8 @@ PACKED_STRUCT_UNALIGNED(struct rb_io_buffer_t {
 typedef struct rb_io_buffer_t rb_io_buffer_t;
 
 typedef struct rb_io_t {
+    VALUE self;
```

Comment:
Just for reference, this is a temporary hack.

Reply:
It turned out to be useful in the `rb_mutex_struct` too, and a common pattern, e.g. `rb_fiber_struct`.